### PR TITLE
Feature/flw 1853 add sub assets for karura

### PR DESCRIPTION
**Pull request by DRadmir**

### CODE REVIEW

File: fearless/Common/Configs/ApplicationConfigs.swift [hunk 1]
```diff
@@ -167,7 +167,11 @@ extension ApplicationConfig: ApplicationConfigProtocol {
     }
 
     var assetListURL: URL? {
-        GitHubUrl.url(suffix: "chains/assets.json")
```
Comment (bnsports) [inline fearless/Common/Configs/ApplicationConfigs.swift:170]: move it to separate PR please

File: fearless/Common/Configs/ApplicationConfigs.swift [hunk 2]
```diff
@@ -192,7 +196,7 @@ private enum GitHubUrl {
         URL(string: "https://raw.githubusercontent.com/soramitsu/fearless-utils/")
     }
 
-    private static let defaultBranch = "ios/v2.0.10"
+    private static let defaultBranch = "ios/v3-dev"
```
Comment (bnsports) [inline fearless/Common/Configs/ApplicationConfigs.swift:199]: why v3-dev ? we have *_dev files for those purposes

File: fearless/Common/DataProvider/ExistentialDeposit.swift [hunk 3]
```diff
@@ -0,0 +1,125 @@
+import Foundation
+import BigInt
+import RobinHood
+import FearlessUtils
+
+protocol ExistentialDepositServiceProtocol {
+    func fetchExistentialDeposit(completion: @escaping (Result<BigUInt, Error>) -> Void)
+}
+
+enum ExistentialDepositCurrencyId {
```
Comment (bnsports) [inline fearless/Common/DataProvider/ExistentialDeposit.swift:10]: I'd prefer to store model in separate file

File: fearless/Common/DataProvider/ExistentialDeposit.swift [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    private let chainAsset: ChainAsset
+    private let runtimeCodingService: RuntimeCodingServiceProtocol
+    private let operationManager: OperationManagerProtocol
+    private let engine: JSONRPCEngine
+
+    // MARK: - Constructor
+
+    init(
+        chainAsset: ChainAsset,
```
Comment (bnsports) [inline fearless/Common/DataProvider/ExistentialDeposit.swift:65]: please do not store chainAsset, it makes service chainAsset-based, while it should be universal

File: fearless/Common/Services/RemoteSubscription/WalletRemoteSubscriptionService.swift [hunk 5]
```diff
@@ -20,26 +19,29 @@ protocol WalletRemoteSubscriptionServiceProtocol {
...
         of accountId: AccountId,
-        chain: ChainModel,
+        chainAsset: ChainAsset,
         queue: DispatchQueue?,
         closure: RemoteSubscriptionClosure?
     ) -> UUID? {
         do {
-            let storagePath = chain.isOrml ? StorageCodingPath.tokens : StorageCodingPath.account
+            var storagePath: StorageCodingPath
+            switch chainAsset.chainAssetType {
```
Comment (bnsports) [inline fearless/Common/Services/RemoteSubscription/WalletRemoteSubscriptionService.swift:28]: it will be move universal if you move this logic to ChainAsset extension (e.g. ChainAsset+Palletes). all necessary data stored inside ChainAsset. this way you can avoid code duplicates in case you need to decide what pallete to use in other modules.

File: fearless/Common/Storage/EntityToModel/AssetModelMapper.swift [hunk 6]
```diff
@@ -10,17 +10,27 @@ final class AssetModelMapper: CoreDataMapperProtocol {
     var entityIdentifierFieldName: String { "id" }
 
     func transform(entity: CDAsset) throws -> AssetModel {
-        guard let id = entity.id, let chainId = entity.chainId else {
+        guard
+            let id = entity.id,
+            let chainId = entity.chainId,
+            let symbol = entity.symbol,
+            let typeRawValue = entity.type
```
Comment (bnsports) [inline fearless/Common/Storage/EntityToModel/AssetModelMapper.swift:17]: in AssetModel constructor you try to create ChainAssetType for given typeRawValue and provide default value (.normal) in case of fail. so it looks like optional field

**Review by Viscount88 — APPROVED** [review]
